### PR TITLE
Realtime CDR Generation Algorithm

### PR DESCRIPTION
**Pull request by dinushadck**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -0,0 +1,476 @@
...
+
+
+            var cdrSave = dbModel.CallCDR.build(cdr);
+
+
+            backendHandler.AddCDRRecord(cdrSave, function(err, result)
+            {
+                //Add to Queue
+
+                if(cdr.Direction === 'inbound' && cdr.ObjCategory !== 'CONFERENCE' && (cdr.OriginatedLegs !== null || (cdr.OriginatedLegs === null && (cdr.ObjType === 'HTTAPI' || cdr.ObjType === 'SOCKET' || cdr.ObjType === 'REJECTED' || cdr.ObjType === 'FAX_INBOUND' || cdr.ObjCategory === 'DND' || cdr.ObjCategory === 'OUTBOUND_DENIED'))))
```
Comment (sukitha) [inline app.js]: move object type in to an array and check for array index

File: app.js [hunk 2]
```diff
@@ -0,0 +1,476 @@
...
+                {
+                    //add to queue
+                    cdr.TryCount = 0;
+
+                    console.log('================== NEW A LEG PUBLISH TO QUEUE - UUID : ' + cdr.Uuid + ' ==================');
```
Comment (sukitha) [inline app.js]: remove console logs and use common logger

File: app.js [hunk 3]
```diff
@@ -0,0 +1,476 @@
...
+    catch(ex)
+    {
+        logger.error('[DVP-CDREventListner.ProcessCDR] - [%s] - Exception occurred', reqId, ex);
+        res.end("{}");
+    }
+
+    return next();
+});
+
+function Crossdomain(req,res,next){
```
Comment (sukitha) [inline app.js]: not require if using cors library

File: app.js [hunk 4]
```diff
@@ -0,0 +1,476 @@
+var restify = require('restify');
+var backendHandler = require('./DBBackendHandler.js');
+var dbModel = require('dvp-dbmodels');
+var config = require('config');
+var nodeUuid = require('node-uuid');
```
[the remark this answers is not in the document]
Comment (dinushadck) [reply app.js]: node uuid package is removed and uuid module is now added.

File: app.js [hunk 5]
```diff
@@ -0,0 +1,476 @@
...
+
+
+server.post('/DVP/API/:version/CDREventListner/ProcessCDR', function(req,res,next)
+{
+    var reqId = nodeUuid.v1();
+
+    try
+    {
+        logger.info('[DVP-CDREventListner.ProcessCDR] - [%s] - FS CDR Request Received', reqId);
+        var cdrObj = req.body;
```
[the remark this answers is not in the document]
Comment (dinushadck) [reply app.js]: req.body is checked for null reference - DONE

File: app.js (hunk 1 above)
Comment (dinushadck) [reply app.js]: moved to an array

var arr = ['HTTAPI', 'SOCKET', 'REJECTED', 'FAX_INBOUND'];

File: app.js (hunk 2 above)
Comment (dinushadck) [reply app.js]: commpn logger added in place of all console logs

File: app.js (hunk 3 above)
Comment (dinushadck) [reply app.js]: removed crossdomain and client access policy functions

**Pull request merged**